packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1847,9 +1847,18 @@ async function renderToStream(
     // one task before continuing
     await waitAtLeastOneReactRenderTask()
 
+    // When streaming metadata is enabled and request UA is a html-limited bot, we should do a dynamic render.
+    const shouldDoDynamicRender =
+      ctx.renderOpts.botType === 'html-limited' &&
+      ctx.renderOpts.experimental.streamingMetadata
+
     // If provided, the postpone state should be parsed as JSON so it can be
     // provided to React.
-    if (typeof renderOpts.postponed === 'string') {
+    if (
+      typeof renderOpts.postponed === 'string' &&
```

Comment:
I think actually the factoring here is wrong. If we are ever in a position to do a render and have a postponed state we should already know that we're not serving an html-limited bot. The reason is that we can't know for certain whether the partial prerender was or was not sent, that's the job of the router. So if we are going to skip the partial prerender b/c we're serving an html limited bot then we ought to not send along the postponed state and can thus keep the original logic here unchanged

Reply:
Addressed in https://github.com/vercel/next.js/pull/75366/commits/06e4fed82701588140cdb10b492855aa3b73351c